### ISSUE TITLE
Add-on: AutoDNC 

### ISSUE THREAD

**Issue by Ghostthief** [Ivaar/Windower-addons#29]
Add-on: AutoDNC getting stuck near "no foot rise" section keeps saying "Unable to use ability"
"Already reached max finishing moves"
 then repeats "feather step" and Presto never does flourish to WS

Seems like a great Add-on if cleaned up sadly I don't have the skill to.

Please take a look.

**Comment by Ivaar** [comment]
Hmmm, sounds like your gearswap is cancelling it and printing that message. It should use CF before WS, provided you have a step available. Its also designed to spam steps/flourishes on cooldown for auto-dynamis staggering. That said I now realize my local copy is different and fixes proc commands. I will push those changes now. Before I go dust of DNC, can you confirm you were at your max steps when receiving that message?

**Comment by Ivaar** [comment]
Ah, I see that system message, however it did CF prior to ws when I had step available. Do //dnc active and make sure ja and ws is set to on, also confirm min/max ws hp and distance.

**Comment by Ivaar** [comment]
This is due to my using gain/lose buff event, which is a bit slower than checking buffs in memory or watching packets (if my memory serves). It doesnt set stuck on that line and moves to the following ability on next iteration. So I do not think that is the issue.

**Comment by Ivaar** [comment]
If it doesn't use ws at all I would have to guess its the distance check. I could enhance that by calculating it in euclidean space, however the exact formula used by the game is currently unknown.

**Comment by Ivaar** [comment]
This should fix the "unable to use" message, would you mind trying it out before I push changes?
<https://gist.github.com/Ivaar/a2da876b0f05f09a666b6dd34d98b358>

It still spams enemy target job abilities when JA mode is enabled and it could waltz right after your CF if your hp drops below your configured threshold. 

If you plan to use JA mode outside dyna you may want to modify that logic so it is not too spammy. I suppose I could implement a second JA mode, to preserve the logic behind stagger mode.

**Comment by Ghostthief** [comment]
On my way home now to try it .... Yes I had finishing move 5 (even though DNC is mastered)

**Comment by Ghostthief** [comment]
Just asking but does waltzpt section function?

**Comment by Ghostthief** [comment]
//dnc ws on made it fire ws but it never used climactic or reverse or wild for me

**Comment by Ghostthief** [comment]
This was on old version not the new one still driving home

**Comment by Ivaar** [comment]
Party waltzes should work, the setting to disable party slots was simply never implemented.
Idk why party waltzes are more comprehensive, covering every tier. I was using this with motes gearswap libs back then change to appropriate waltz. That said self/party waltz could be combined into the waltz settings, maybe that's where I was going with it.

The buff adjustments I just made may possibly resolve your issue.

**Comment by Ghostthief** [comment]
getting ready to test now

**Comment by Ghostthief** [comment]
ok the tests worked .... But I can't turn waltzpt off

**Comment by Ivaar** [comment]
um, I believe it just needs to be added to dual_toggle

**Comment by Ghostthief** [comment]
Yep that fixed it thanks

**Comment by Ghostthief** [comment]
What is ws2 "Cyclone" used for?

**Comment by Ivaar** [comment]
That was going to be for ws stagger mode, never cared enough to flesh that out.

**Comment by Ghostthief** [comment]
Hmmm wonder If I am smart enough to make it into a self skillchain mode

**Comment by Ghostthief** [comment]
What could I do to send an echo of each event to the other chat window .... Hard to follow when in combat scroll

**Comment by Ivaar** [comment]
Run this script to find a chat mode (color) that prints to desired log window <https://github.com/Windower/Lua/blob/dev/scripts/colortest.lua>
in game settings affect what modes are displayed in which log i.e. config>log>window X>
and config>chat filters can block modes from displaying.

What do you mean each event?

**Comment by Ghostthief** [comment]
the flourishes and such

**Comment by Ghostthief** [comment]
step also
 go by so fast

**Comment by Ivaar** [comment]
I had this laying around, care to see if it works? <https://gist.github.com/Ivaar/a5bd990110b4b12a21298fd0db789a72>

**Comment by Ghostthief** [comment]
I'll give it a go in a few sure

**Comment by Ghostthief** [comment]
I loaded it but not sure what it's suppose to do ... It didn't fire anything that I saw

**Comment by Ivaar** [comment]
It creates a text box with step levels for current target, looks like it prints an addon message as well (likely for debugging). Doesn't track or log flourishes.

**Comment by Ghostthief** [comment]
I'll try it again in the morning when I'm off work but I saw no text box

**Comment by Ghostthief** [comment]
Okay I see it now YES it tracks steps and the lvl on them ... but I changed to X,Y in the settings xml file but it still puts it at 0,0

**Comment by Ghostthief** [comment]
I figured it out
Edited x,y in actual LUA and it moved it... only problem is it doesn't seem to register when the debuff drops

**Comment by Ivaar** [comment]
I looked at it the other day and noticed it was unfinished and does not clear the effects properly.
You can check out the debuffs addon to see if that has similar functionality. If not, I can possibly look into it this week.

**Comment by Ghostthief** [comment]
also noticed it doesn't track Featherstep
only quickstep, Box step and Stutter step

**Comment by Ghostthief** [comment]
step_message = {
    [519] = 'Lethargic Daze',
    [520] = 'Sluggish Daze',
    [521] = 'Weakened Daze',
    [591] = 'Bewildered Daze',
    }
that should fix that minor problem

**Comment by Ghostthief** [comment]
well think it will need more at the bottom but yeah

**Comment by Ivaar** [comment]
I updated steps.lua, misc issues should be resolved and it now displays a timer and if you are the actor it will account for job points.

**Comment by Ghostthief** [comment]
oh nice this works great

**Comment by Ghostthief** [comment]
thanks a bunch
you got AutoDNC working awesome and now Steps is a second addition to your awsomeness!
